### ISSUE TITLE
MO labels with "_"

### ISSUE THREAD

**Issue by vpohl** [orbkit/orbkit#17]
If I have a molecular orbital label with an underscore as in Gunter's NTO program, MO selection with integer or boolean arrays fails due to `mo_spec.get_spinstate()`.

```
 >>> qc.mo_spec.get_sym()[1]
'330.NTO_h'
>>> qc.mo_spec[[1]]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/spetry/orbkit/orbkit/orbitals.py", line 490, in __getitem__
    mo_out.update()
  File "/home/spetry/orbkit/orbkit/orbitals.py", line 604, in update
    self.get_spinstate()
  File "/home/spetry/orbkit/orbkit/orbitals.py", line 639, in get_spinstate
    spindic[split_label[-1]].append(isym)
KeyError: 'h'
```

**Comment by lukasmarsoner** [comment]
Are there no spin-labels on the NTOs? Anyway. Though I think one can probably fix the problem for now by checking for hole/particle labels explicitly, this whole get_spinstate() business is just bad design at this point. This was a somewhat lazy fix for the fact that we did not use to have a way to keep track of the spin-state of MOs even after the MOClass was introduced.

Now that we handle spins properly via self.alpha_index/self.beta_index there is no need to continue this madness. The best way to solve this would be to just generate lists of alpha and beta orbital indexes in the readers directly. This could be done quite easily by just inserting appends at those points where _a/_b is added to the orbital labels.

I'm not sure if I can get to it right away though...

**Comment by vpohl** [comment]
What do you think? Wouldn't it be better and simpler, if we just introduce a 'spin' item in the mo_spec dictionary? So we can stick to the spirit of the user/function being able to access and modify all variables of mo_spec as if it would be just a list of dictionaries...

**Comment by lukasmarsoner** [comment]
I think that would go on top of what i proposed but yea - what we should do is just add the two spin-index arrays to the dictionary. Maybe also a Boolean telling the user directly if the calculation is spin-polarized or not. The latter is kind of redundant but still nice to have I think.

**Comment by vpohl** [comment]
Can someone please check the implementation of the spin into qc.mo_spec? What do you think? It should also fix Issue #18. 

At the momement, the spin is either 'alpha' or 'beta', should we add 'paired' or 'alpha_beta' or should we just leave it like it is?
